web/ui_utils.js

```diff
@@ -994,6 +997,29 @@ function apiPageModeToSidebarView(mode) {
   return SidebarView.NONE; // Default value.
 }
 
+function getXfaHtmlForPrinting(printContainer, pdfDocument) {
+  const xfaHtml = pdfDocument.allXfaHtml;
+  const factory = new DefaultXfaLayerFactory();
+  const scale = Math.round(CSS_UNITS * 100) / 100;
+  for (const xfaPage of xfaHtml.children) {
+    const page = document.createElement("div");
+    page.setAttribute("class", "xfaPrintedPage");
+    printContainer.appendChild(page);
+
+    const { width, height } = xfaPage.attributes.style;
+    const viewBox = [0, 0, parseInt(width), parseInt(height)];
+    const viewport = new PageViewport({ viewBox, scale, rotation: 0 });
```

Comment:
Can we pretty please add an API method which returns the necessary `PageViewport` and use that instead here, rather than exposing this class directly in the *public API*?

Unless absolutely necessary, this class isn't the sort of thing that we ideally want to expose publicly since it makes it more difficult to e.g. make any future changes to that code.